core/src/main/java/org/apache/calcite/sql/SqlAlterProcedure.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+
+/**
+ * Parse tree for {@code ALTER PROCEDURE} statement.
+ */
+public class SqlAlterProcedure extends SqlAlter {
+  public final SqlIdentifier procedureName;
+  public final boolean languageSql;
+  public final List<AlterProcedureWithOption> options;
+  public final boolean local;
+  public final boolean isTimeZoneNegative;
+  public final String timeZoneString;
+
+  private static final SqlOperator OPERATOR =
+      new SqlSpecialOperator("ALTER PROCEDURE", SqlKind.ALTER_PROCEDURE);
+
+  /**
+   * Creates an instance of {@code SqlAlterProcedure}.
+   *
+   * @param pos The parser position
+   * @param procedureName Name of the procedure
+   * @param languageSql If LANGUAGE SQL is specified
+   * @param options List of WITH options
+   * @param local If LOCAL was specified for AT TIME ZONE
+   * @param isTimeZoneNegative True if sign before timeZoneString is "-"
+   * @param timeZoneString String after AT TIME ZONE, may be null
+   */
+  public SqlAlterProcedure(SqlParserPos pos, String scope,
+      SqlIdentifier procedureName, boolean languageSql,
+      List<AlterProcedureWithOption> options, boolean local,
+      boolean isTimeZoneNegative, String timeZoneString) {
+    super(pos, scope);
+    this.procedureName = procedureName;
+    this.languageSql = languageSql;
+    this.options = options;
+    this.local = local;
+    this.isTimeZoneNegative = isTimeZoneNegative;
+    this.timeZoneString = timeZoneString;
+  }
+
+  @Override public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
+    writer.keyword("ALTER PROCEDURE");
+    unparseAlterOperation(writer, leftPrec, rightPrec);
+  }
+
+  @Override protected void unparseAlterOperation(SqlWriter writer, int leftPrec,
+      int rightPrec) {
+    procedureName.unparse(writer, 0, 0);
+    if (languageSql) {
+      writer.keyword("LANGUAGE SQL");
+    }
+    writer.keyword("COMPILE");
+    if (!options.isEmpty()) {
```

Comment:
This could throw a NullPointerException if options is null - might be worth adding an Objects.requireNonNull in constructor for this. Also applies to procedureName.

Reply:
Yeah that's a good point, I updated them with requireNonNull calls